packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -835,20 +835,51 @@ export class SequenceMode extends BaseMode {
     const chainsCollection = ChainsCollection.fromMonomers([
       ...drawingEntitiesManager.monomers.values(),
     ]);
+
+    const currentSequence = SequenceRenderer.currentChain;
+
+    const currentSequenceHasPhosphate = currentSequence?.lastNonEmptyNode
+      ? this.hasPhosphateAtEndOfSequence(
+          currentSequence.lastNonEmptyNode.monomer,
+        )
+      : false;
+
+    const newSequenceHasPhosphate = chainsCollection.lastNode
+      ? this.hasPhosphateAtEndOfSequence(chainsCollection.lastNode.monomer)
+      : false;
+
+    let nextCaretPosition =
+      SequenceRenderer.caretPosition + chainsCollection.length;
+
+    if (
+      (currentSequenceHasPhosphate && newSequenceHasPhosphate) ||
+      currentSequenceHasPhosphate
+    ) {
+      nextCaretPosition -= 1;
+    }
+
     const modelChanges = this.insertNewSequenceFragment(chainsCollection);
 
     modelChanges.addOperation(new ReinitializeModeOperation());
 
     modelChanges.addOperation(
       new RestoreSequenceCaretPositionOperation(
         SequenceRenderer.caretPosition,
-        SequenceRenderer.caretPosition + chainsCollection.length,
+        nextCaretPosition,
       ),
     );
 
     return modelChanges;
   }
 
+  private hasPhosphateAtEndOfSequence(monomer: BaseMonomer): boolean {
+    if (!monomer) return false;
+    if (!monomer.monomerItem || !monomer.monomerItem.props) {
+      return false;
+    }
+    return monomer.monomerItem.props.Name === 'Phosphate';
```

Comment:
It can be simplified by using conditional chaining: `return monomer?.monomerItem?.props?.Name === 'Phosphate'`, you won't need those conditions above